pennylane/_typing.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This file contains different PennyLane types."""
+import contextlib
+
+# pylint: disable=import-outside-toplevel, too-few-public-methods
+import sys
+from typing import Union
+
+import numpy as np
+from autograd.numpy.numpy_boxes import ArrayBox
+
+
+class _Typing:
+    """Class containing useful PennyLane types."""
+
+    @property
+    def TensorLike(self):
```

Comment:
Might be worth turning this into a cached property somehow so we don't have to recalculate it each time.

Reply:
The idea is to recalculate it each time. Otherwise if the user import one of the packages (`torch`, `tensorflow`...) after importing `pennylane`, its related tensor-like objects won't be taken into account.